scripts/sv_logs.lua

```diff
@@ -0,0 +1,111 @@
+if GetConvar('txAdminServerMode', 'false') == 'true' then
+	-- micro optimization
+	local os_time = os.time
+	-- http://lua-users.org/wiki/SimpleRound
+	local function round(num)
+		return tonumber(string.format("%.2f", num))
+	end
+
+	local function getPlayerData(src)
+		if not src then return false end
+		if src == -1 then return {name = 'console', identifiers = {}} end
+		return {
+			name = GetPlayerName(src),
+			identifiers = GetPlayerIdentifiers(src)
+		}
+	end
+
+	local apiPort = GetConvar("txAdmin-apiPort", "invalid")
+	local apiToken = GetConvar("txAdmin-apiToken", "invalid")
+	if apiPort == "invalid" or apiToken == "invalid" then
+		logError('API Port and Token ConVars not found. Do not start this resource if not using txAdmin.')
+		return
+	end
+
+	-- back ticks break my intellisense
+	local PRINT_STRUCTURED_TRACE = GetHashKey('PRINT_STRUCTURED_TRACE')
+	local function log(src, action, data)
```

Comment:
`log` and `logError` words should be used for terminal logging, like in `sv_main.lua`.
Probably better to make those print functions a separate file and include here.  
  
So, for the actual fd3 log function, use another name.  
Also to prevent spamming and broken pipes, add all to a buffer and flush it every second or so. Maybe even 2~3.

Reply:
> So, for the actual fd3 log function, use another name.
> Also to prevent spamming and broken pipes, add all to a buffer and flush it every second or so. Maybe even 2~3.

Would this really be necessary? It was something that was in the initial concept when it was initially using HTTP instead of fd3, but it complicated the code a lot and I didn't see it as something worth having, though this could be due to my lack of knowledge of how txAdmin and fd3 in general works.